packages/amesos2/src/Amesos2_TpetraMultiVecAdapter_def.hpp

```diff
@@ -143,8 +144,51 @@ namespace Amesos2 {
     // Redistribute the input (multi)vector.
     redist_mv.doExport (*mv_, *exporter_, Tpetra::REPLACE);
 
+    // Do this if GIDs not contiguous...
+    // getLocalView, copy into Teuchos::ArrayView av
+    // Need safety about the sizes...
+    typedef typename multivec_t::dual_view_type dual_view_type;
+    typedef typename dual_view_type::host_mirror_space host_execution_space;
+    redist_mv.template sync < host_execution_space > ();
+    
+    //auto myRank = distMap->getComm()->getRank();
+    auto contig_local_view_2d = redist_mv.template getLocalView<host_execution_space>();
```

Comment:
@ndellingwood It's actually incorrect if the MultiVector is noncontiguous (`X.isConstantStride() == false`).  The 2-D View points to the original (parent) MultiVector, not to its child.  You could just call ```MultiVector::get1dCopy```, or do this:
```
if (X.isConstantStride ()) {
  // ... just do what you're doing ... 
}
else {
  // ... lda should come from Teuchos::Array* allocation,
  // not the MultiVector, since the MultiVector does NOT
  // have constant stride in this case.
  const size_t lclNumRows = X.getLocalLength ();
  for (size_t j = 0; j < X.getNumVectors (); ++j) {
    auto av_j = av (lda*j, lclNumRows);
    auto X_j = X.getVector ();
    auto X_lcl_j_2d = X.template getLocalView<host_execution_space> ();
    auto X_lcl_j_1d = Kokkos::subview (X_lcl_j_2d, Kokkos::ALL (), 0);
    // ... copy from av_j into X_lcl_j_1d, a 1-D Kokkos::View, as before ...
  }
}
```